content/backend/typescript-apollo/0-introduction.md

```diff
@@ -0,0 +1,130 @@
+- Some experience working with databases (we're going to use a relational database but experience with document databases like MongoDB works too). 
+
+
+##### Assumed Tools and setup
+
+Throughout this tutorial, we'll be making a few minor assumptions about your tools you are using: 
+
+- npm as the package manager.
+- VSCode as the IDE. 
+- `Unix` shell (like the terminal/shell in `linux` and `macOS`).
+
+If you're using another set of tools, like Yarn (package manager), Windows (OS) or Webstorm (IDE), that's totally fine and welcome! Just know that you'll occasionally need to adapt minor instructions to your situation.
+
+##### Annotated code blocks
+
+Most Code Blocks are explained with annotations. Take the following for example. Immediately following the codeblock are numbered points. Each corresponds to a line annotated with that number in the code block.
+
+```typescript
+const a = 1          // 1
+const b: any = 1     // 2
+// @ts-ignore        // 3
+const c: number = "Not a number";
+```
+1. Typescript automatically infers the type of A `number` since it is not provided. 
+2. Use `any` to effectively disable type-checking and allow any type. 
+3. Use `// @ts-ignore` before a line to ignore typescript compiler errors. 
+
+##### Auxiliary content
+
+Sometimes **Note**s will be used to highlight content that isn't core to the current flow but related. It looks like this: 
+
+> **Note:** If you're interested to learn more about `abc`, you should check out `xyz` resource. 
+
+
+##### Package installation with fixed major version
+
+_Most_ package installations throughout the tutorial will follow a format similar to this:
+
+```bash
+npm install package_name@^version 
+```
+
+The `^` indicates to npm to keep the major version intact and update to latest minor and patch version when installing the package. According to [semver rules](https://semver.org/) breaking changes should only be introduced during major version upgrades. Keeping the major version fixed like this ensures there are no breaking changes in any of the packages you install that _unintentionally breaks the tutorial_. 
+
+If you want to upgrade a package to a new major version not instructed in the tutorial, please make sure you know what you are doing. 
+
+
+
+##### Copy-pasting code from the tutorial
+
+As you progress you'll be writing code, of course. Often you'll have the chance to copy and paste our code, but we strongly suggest if you're just starting out to **write the code yourself**. This will expose you to the _experience_ of writing code in a Typescript codebase with type-safety features. This includes things like autocompletion and letting static safety guide your implementations, e.g. inside a GraphQL resolver. Also if you're new to TypeScript getting used to encountering, understanding, and fixing static type errors is essential practice. 
```

Comment:
> This will expose you to the _experience_ of writing code in a Typescript codebase with type-safety features. This includes things like autocompletion and letting static safety guide your implementations,

Really like this note about why it's beneficial to type things out with TypeScript!

Reply:
Thank you. I took inspiration from the [nexus docs](https://nexusjs.org/docs/getting-started/tutorial/chapter-introduction) has a similar note :)